### PR TITLE
Discord Suppress Notifications on send message actions

### PR DESCRIPTION
**Pull request by rhombuslabs**
## WHAT

copilot:summary

copilot:poem

## WHY

About a year ago, Discord added the ability to send silent messages which trigger no notifications in channels. The equivalent for bots was a new flag SUPPRESS_NOTIFICATIONS added to sendMessage(). I've added a boolean parameter, defaulting to false, to take advantage of this. Here's a link to the Discord documentation on it:

https://github.com/discord/discord-api-docs/blob/main/docs/resources/Channel.md#message-flags

and the commit for that markdown, with an explainer from the contributor

https://github.com/discord/discord-api-docs/commit/b7d6467954ab7feb61f1bb749284c61165db6a99#diff-160acb4654c611c67c4ee46b2ce5dda2be82212af8d1d00cd0256ef6edda1e95

## HOW

copilot:walkthrough

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/discord/actions/common/common.mjs [hunk 1]
```diff
@@ -61,5 +67,13 @@ export default {
       const workflowId = process.env.PIPEDREAM_WORKFLOW_ID;
       return `Sent via [Pipedream](<https://pipedream.com/@/${workflowId}?o=a&a=discord>)`;
     },
+    getMessageFlags(suppressNotifications) {
+      const SUPPRESS_NOTIFICATIONS = 4096;
+
+      var flags = 0;
+      if (suppressNotifications) flags += SUPPRESS_NOTIFICATIONS;
+
+      return flags;
```
Comment (lcaresia) [inline components/discord/actions/common/common.mjs:76]: Please, move this to a `constants.mjs` file

```suggestion
      var flags = 0;
      if (suppressNotifications) flags += constants.SUPPRESS_NOTIFICATIONS;

      return flags;
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Just need one fix, thanks for your contribution.

File: components/discord/actions/common/common.mjs (hunk 1 above)
Comment (rhombuslabs) [reply components/discord/actions/common/common.mjs]: Done.

**Review by lcaresia — APPROVED** [review]

**Comment by rhombuslabs** [comment]
@lcaresia Am I responsible for merging pnpm-lock.yaml?

**Comment by rhombuslabs** [comment]
I guess I keep having to resolve pnpm merge conflicts until this PR is committed?

**Comment by rhombuslabs** [comment]
> Hello everyone, I have tested this PR and there're some test cases failed or needed improvement.
> 
> Please check the test report below for more information https://vunguyenhung.notion.site/Discord-Suppress-Notifications-on-send-message-actions-10946-3d4bdee8416e48fcbfdedfbb9b6a55c1

I'm not certain what "test cases" there are, there's none in the code that I see, would appreciate it if you would point me to them.

Also, I would recommend attempting to publish again, perhaps that's a temporary problem. I was able to publish to Pipedream and test, no problems.

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**